### PR TITLE
Extract st terminal font from Xresources

### PR DESCRIPTION
**Pull request by AtifChy**
## Description
Extracts font set for st terminal from xresources. st terminal needs to patched with [the xresource patch](https://st.suckless.org/patches/xresources/)

## Features
xresources patch support
font2 patch support

### CODE REVIEW

File: neofetch [hunk 1]
```diff
@@ -3549,16 +3549,26 @@ END
...
+                    xrdb=$(xrdb -query)
+                    term_font=$(grep -im 1 -e "^${term/d}"'\**\.*font:' -e '^\*font:' <<< "$xrdb")
+                    term_font=${term_font/*"*font:"}
+                    term_font=${term_font/*".font:"}
+                    term_font=${term_font/*"*.font:"}
+                    term_font=$(trim "$term_font")
+                fi
+
+                if [[ -z $term_font && -n "$binary" ]]; then
+                    term_font=$(strings "$binary" | tac | \
```
Comment (dylanaraps) [inline neofetch:3562]: Why is the `tac` call needed? 

Also, it's better to use something more portable like awk: 

```sh
awk '{a[i++]=$0} END {for (j=i-1; j>=0;) print a[j--] }'
```

File: neofetch [hunk 2]
```diff
@@ -3549,16 +3549,26 @@ END
                 # like a font definition. NOTE: There is a slight limitation in this approach.
                 # Technically "Font Name" is a valid font. As it doesn't specify any font options
                 # though it is hard to match it correctly amongst the rest of the noise.
-                [[ -n "$binary" ]] &&
-                    term_font=$(
-                        strings "$binary" |
+                if [[ -n $(strings "$binary" | grep resource) ]]; then
```
Comment (dylanaraps) [inline neofetch:3552]: Can we save `strings "$binary"` to a variable so we don't call it twice in some cases?

File: neofetch [hunk 3]
```diff
@@ -3549,16 +3549,26 @@ END
...
+                    xrdb=$(xrdb -query)
+                    term_font=$(grep -im 1 -e "^${term/d}"'\**\.*font:' -e '^\*font:' <<< "$xrdb")
+                    term_font=${term_font/*"*font:"}
+                    term_font=${term_font/*".font:"}
+                    term_font=${term_font/*"*.font:"}
+                    term_font=$(trim "$term_font")
+                fi
+
+                if [[ -z $term_font && -n "$binary" ]]; then
```
Comment (dylanaraps) [inline neofetch:3561]: Why do we check if `$binary` is non-null here? (`strings` above runs with no check).

File: neofetch (hunk 2 above)
Comment (dylanaraps) [inline neofetch:3552]: Grep call should use `-F` as we're sending a plain string.

Comment (dylanaraps) [reply neofetch:3552]: This whole thing would be better with:

```sh
st_bin=$(strings "$binary")

if [[ $st_bin == *resource* ]]; then
    # ...
fi 

# ...

File: neofetch (hunk 3 above)
Comment (AtifChy) [reply neofetch:3561]: 3561-3568 is for font2 patch

File: neofetch [hunk 4]
```diff
@@ -3549,16 +3549,26 @@ END
...
                 # though it is hard to match it correctly amongst the rest of the noise.
-                [[ -n "$binary" ]] &&
-                    term_font=$(
-                        strings "$binary" |
+                if [[ -n $(strings "$binary" | grep resource) ]]; then
+                    xrdb=$(xrdb -query)
+                    term_font=$(grep -im 1 -e "^${term/d}"'\**\.*font:' -e '^\*font:' <<< "$xrdb")
+                    term_font=${term_font/*"*font:"}
+                    term_font=${term_font/*".font:"}
+                    term_font=${term_font/*"*.font:"}
```
Comment (dylanaraps) [inline neofetch:3557]: Does this work to replace all of these lines?

```
term_font=${term_font/*[\*.]"font:"}
```

File: neofetch [hunk 5]
```diff
@@ -3549,16 +3549,26 @@ END
                 # like a font definition. NOTE: There is a slight limitation in this approach.
                 # Technically "Font Name" is a valid font. As it doesn't specify any font options
                 # though it is hard to match it correctly amongst the rest of the noise.
-                [[ -n "$binary" ]] &&
-                    term_font=$(
-                        strings "$binary" |
+                if [[ -n $(strings "$binary" | grep resource) ]]; then
+                    xrdb=$(xrdb -query)
```
Comment (dylanaraps) [inline neofetch:3553]: `xrdb` is only used once (following line). Why not just do:

```
term_font=$(xrdb -query | grep -im 1 -e "^${term/d}"'\**\.*font:' -e '^\*font:')
```

File: neofetch (hunk 3 above)
Comment (dylanaraps) [reply neofetch:3561]: Yes, that's not what I asked.

**Pull request closed without merging**